File: striver_a2z/src/twoSum.java
import java.util.HashMap;
import java.util.HashSet;

public class twoSum {
    public static void main(String[] args) {

    }

    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] ans = new int[2];
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(target-nums[i])){
                ans[0] = nums[i];
                ans[1] = map.get(target-nums[i]);return ans;
            }
            else{
                map.put(nums[i], i);
            }
        }return ans;
    }
}

File: neetcode_150/twoSum.py
from typing import List
def twoSum(self, nums: List[int], target: int) -> List[int]:
    values = dict()
    for i in range(nums):
        if target-nums[i] in values:
            return [values.get(target-nums[i]), i]
        else:
            values[nums[i]] = i
    return [0,0]